fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    this.getAllTodos();
+  }
+
+  getAllTodos() {
+    fetch('/api/todos')
+      .then(response => {
+        if (!response.ok) {
+          throw new Error('Fetch request has failed');
+        }
+        return response;
+      })
+      .then(response => response.json())
+      .then(data => {
+        this.setState({ todos: data });
+      })
+      .catch(error => console.error(error));
+
+  }
+
+  addTodo(newTodo) {
+    fetch('/api/todos', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(newTodo)
+    })
+      .then(response => response.json())
+      .then(data => {
+        const updatedTodos = this.state.todos;
+        updatedTodos.push(data);
```

Comment:
Be sure to make a copy of the array first. `Array.prototype.slice` can be used for that, but assigning a reference data type to another variable is not a copy.